### PR TITLE
Fix redundant linkage with boost in mailio

### PR DESCRIPTION
**Pull request by Fefer-Ivan**
Specify library name and version:  **mailio/0.20.0**

Because in package_info there is no specific requirements for boost components, in generated files, mailio links with whole boost. In some cases it can create troubles, so I want to fix it.

Also, when building the library, it also linked with whole boost because it used cmake generator and CONAN_LIBS.

Thanks.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Fefer-Ivan** [comment]
It looks like error fixed by this PR: https://github.com/conan-io/conan-center-index/pull/4597
Also it looks like this PR is not propagated to conan-center repository yet, so all I need to do is wait.

File: recipes/mailio/all/conanfile.py [hunk 1]
```diff
@@ -20,9 +20,9 @@ class mailioConan(ConanFile):
         "fPIC": True,
         "shared": False
     }
-    requires = ["boost/1.75.0", "openssl/1.1.1i"]
-    generators = "cmake"
-    exports_sources = ["CMakeLists.txt", "patches/**"]
+    requires = ["boost/1.75.0", "openssl/1.1.1j"]
+    generators = "cmake", "cmake_find_package"
```
Comment (mathbunnyru) [inline recipes/mailio/all/conanfile.py:24]: You probably don't need cmake generator here and can simplify CMakeLists.txt wrapper as well.
Or even get rid of it.

**Review by mathbunnyru — APPROVED** [review]

File: recipes/mailio/all/conanfile.py (hunk 1 above)
Comment (Fefer-Ivan) [reply recipes/mailio/all/conanfile.py:24]: I still need some way to add build dir to CMAKE_MODULE_PATH so find_package can find generated Find<lib>.cmake.

1. One way to it is include cmake generator and call conan_basic_setup().
2. Another way is to include cmake_paths generator and include generated conan_paths.cmake file
 
I searched for cmake_paths line in repo and found none, so I went with the first option.

**Review by mathbunnyru — DISMISSED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
#4597 make this finally possible

 💟 @SpaceIm

File: recipes/mailio/all/conanfile.py [hunk 2]
```diff
@@ -72,11 +72,12 @@ def configure(self):
         except KeyError:
             self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
 
-    def build(self):
-        patches = self.conan_data["patches"][self.version]
-        for patch in patches:
-            tools.patch(**patch)
+    def _patch_sources(self):
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists, "${CMAKE_BINARY_DIR}/version.hpp", "${PROJECT_BINARY_DIR}/version.hpp")
```
Comment (SpaceIm) [inline recipes/mailio/all/conanfile.py:77]: Shouldn't it be kept in a patch, and submit upstream in a PR?

File: recipes/mailio/all/conanfile.py [hunk 3]
```diff
@@ -88,3 +89,4 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.requires = ["boost::system", "boost::date_time", "boost::regex", "openssl::openssl"]
```
Comment (SpaceIm) [inline recipes/mailio/all/conanfile.py:92]: This feature requires conan 1.30.0
Please add `required_conan_version = ">=1.30.0"` before class declaration

File: recipes/mailio/all/conanfile.py [hunk 4]
```diff
@@ -73,8 +74,9 @@ def configure(self):
             self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
 
     def _patch_sources(self):
-        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        tools.replace_in_file(cmakelists, "${CMAKE_BINARY_DIR}/version.hpp", "${PROJECT_BINARY_DIR}/version.hpp")
+        patches = self.conan_data["patches"][self.version]
+        for patch in patches:
+            tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/mailio/all/conanfile.py]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

**Review by mathbunnyru — DISMISSED** [review]

File: recipes/mailio/all/conanfile.py [hunk 5]
```diff
@@ -72,11 +73,13 @@ def configure(self):
...
             self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
 
-    def build(self):
+    def _patch_sources(self):
         patches = self.conan_data["patches"][self.version]
         for patch in patches:
             tools.patch(**patch)
 
+    def build(self):
+        self._patch_sources()
```
Comment (SpaceIm) [inline recipes/mailio/all/conanfile.py:82]:
```suggestion
    def build(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

Comment (SpaceIm) [reply recipes/mailio/all/conanfile.py]: Allow to mix versions with and without patches

**Review by mathbunnyru — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by mathbunnyru** [comment]
@uilianries could you please take a look? This should be good to go.

**Review by SSE4 — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**